staging/src/k8s.io/apiserver/pkg/storage/etcd3/compact.go

```diff
@@ -230,3 +244,66 @@ func compact(ctx context.Context, client *clientv3.Client, expectVersion, rev in
 	klog.V(4).Infof("etcd: compacted rev (%d), endpoints (%v)", rev, client.Endpoints())
 	return currentVersion, currentRev, rev, nil
 }
+
+func (c *compactor) runWatchLoop(stopCh chan struct{}) {
+	ctx := wait.ContextForChannel(stopCh)
+	for {
+		select {
+		case <-time.After(time.Second):
+		case <-ctx.Done():
+			return
+		}
+		compactRev, currentRev, err := c.getCompactRev(ctx)
+		if err != nil {
+			klog.Errorf("etcd: endpoint (%v): %v", c.client.Endpoints(), err)
+			continue
+		}
+		if compactRev != 0 {
+			c.UpdateCompactRevision(compactRev)
+		}
+		watch := c.client.Watch(ctx, compactRevKey, clientv3.WithRev(currentRev))
+		for resp := range watch {
+			compactRev, err := c.fromWatchResponse(resp)
+			if err != nil {
+				klog.Errorf("etcd: endpoint (%v): %v", c.client.Endpoints(), err)
+				continue
+			}
+			if compactRev != 0 {
+				c.UpdateCompactRevision(compactRev)
+			}
+		}
+	}
+}
+
+func (c *compactor) getCompactRev(ctx context.Context) (compactRev int64, currentRev int64, err error) {
+	resp, err := c.client.Get(ctx, compactRevKey)
+	if err != nil {
+		return compactRev, currentRev, fmt.Errorf("get %q failed: %w", compactRevKey, err)
+	}
+	if len(resp.Kvs) != 0 {
```

Comment:
If this condition evaluates to false, it looks like compactRev will be zero valued for the following code (lines 289-). Is that safe?